talk/basicconcepts/scopesnamespaces.tex

```diff
@@ -31,12 +31,12 @@
 // b[1] = a + 1;
 }
 \begin{frame}[fragile]
-  \frametitlecpp[98]{Scope and lifetime of resources}
-  \begin{block}{Resource life time}
+  \frametitlecpp[98]{Scope and lifetime of variables}
+  \begin{block}{Variable life time}
     \begin{itemize}
-      \item Resources are allocated when declared
-      \item Resources are freed at the end of a scope
-      \item Good practice: initialise resources when allocating!
+      \item Variables are allocated when defined
+      \item Variables are freed at the end of a scope
+      \item Good practice: initialise variables when allocating them!
     \end{itemize}
   \end{block}
```

Comment:
That slide confused me initially. It is the first time we talk about a resource without defining what it is. Then I thought we wanted to talk about RAII. But that only really makes sense after we know about constructors/destructors. So I concluded this slide talks about when variables are allocated and destroyed. So I chose to replace resource by variable. Technically, the term `object` would be more appropriate, but we have a big OOP block later, where the term `object` is then used in the OOP sense.

Reply:
You're absolutely right. And on top I propose to move this part earlier in the talk in #98 so we should stick to simple concepts like variables here.